core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/report/detail/ReportDetailViewModel.kt

```diff
@@ -45,53 +61,106 @@ class ReportDetailViewModel(
         _appUiState.update { prev ->
             prev.copy(
                 fabState =
-                FabUiState(
-                    visible = true,
-                    text = systemImpl.getString(MR.strings.edit),
-                    icon = FabUiState.FabIcon.EDIT,
-                    onClick = this@ReportDetailViewModel::onClickEdit
-                )
+                    FabUiState(
+                        visible = true,
+                        text = systemImpl.getString(MR.strings.edit),
+                        icon = FabUiState.FabIcon.EDIT,
+                        onClick = this@ReportDetailViewModel::onClickEdit
+                    ),
+                title = "Graph",
             )
         }
+        viewModelScope.launch {
+            _uiState.whenSubscribed {
+                launchIfHasPermission(
+                    setLoadingState = true,
+                    permissionCheck = { true }
+                ) {
+                    val siteFlow = activeRepo.reportDao().findByUidLive(reportUid)
+                    launch {
+                        siteFlow.collect { report ->
+                            try {
+                                val reportNonNull = report
+                                    ?: throw IllegalStateException("Report not found for uid $reportUid")
+                                val optionsJson = reportNonNull.reportOptions
+                                val reportTitle = reportNonNull.reportTitle
+
+                                val parsedOptions = when {
+                                    !optionsJson.isNullOrBlank() -> {
+                                        try {
+                                            Json.decodeFromString(
```

Comment:
This should use json not Json so it uses the serializer from dependency injection